File: Shuffle the Array.cpp
class Solution {
public:
    vector<int> shuffle(vector<int>& nums, int n) {
        vector<int> dummy ;

        int i = 0 , j = n , k = 0 ;

        while (k < n) {
            dummy.push_back(nums[i]) ; i++ ;
            dummy.push_back(nums[j]) ; j++ ;
            k++ ;
        }

        return dummy ;
    }
};

File: Shuffle the Array.java
class Solution {
	public int[] shuffle(int[] nums, int n) {
		int[] result = new int[2 * n] ;

		int k = 0 , i = 0 , j = n ;

		while (k < 2 * n) {
			result[k] = nums[i] ; i++ ; k++ ;
			result[k] = nums[j] ; j++ ; k++ ;
		}

		return result ;

	}
}